Jenkinsfile

```diff
@@ -1,35 +1,108 @@
 #!groovy
 
+def artServer = Artifactory.server 'art-p-01'
+def buildInfo = Artifactory.newBuildInfo()
+def distDir = 'build/dist/'
+
 pipeline {
+    libraries {
+        lib('jenkins-pipeline-shared@feature/sbt-config')
+    }
+    agent any
     options {
         skipDefaultCheckout()
-        buildDiscarder(logRotator(numToKeepStr: '30', artifactNumToKeepStr: '30'))
-        timeout(time: 30, unit: 'MINUTES')
-        timestamps()
+        buildDiscarder(logRotator(numToKeepStr: '30', artifactNumToKeepStr: '7'))
+        timeout(time: 10, unit: 'MINUTES')
     }
-    agent any
+    environment {
+        MODULE_NAME = 'sbt-play-defaults'
+    }
+    
     stages {
         stage('Checkout') {
-            agent any
+            agent { label 'download.jenkins.slave' }
             steps {
                 deleteDir()
-                checkout scm
-                stash name: 'app'
-                sh "$SBT version"
+                checkout scm        
+                script {
+                    buildInfo.name = "${MODULE_NAME}"
+                    buildInfo.number = "${BUILD_NUMBER}"
+                    buildInfo.env.collect()
+                }
+                colourText("info","BuildInfo: ${buildInfo.name}-${buildInfo.number}")
+                stash name: 'Checkout'
             }
         }
-
-        stage('Package'){
-            agent any
+        stage('Build'){
+            agent { label 'build.sbt_0-13-13' }
+            steps {
+                unstash name: 'Checkout'
+                sh "sbt compile"
+            }
+            post {
+                success {
+                    colourText("info","Stage: ${env.STAGE_NAME} successful!")
+                }
+                failure {
+                    colourText("warn","Stage: ${env.STAGE_NAME} failed!")
+                }
+            }
+        }
+        stage ('Publish') {
+            agent { label 'build.sbt_0-13-13' }
+            when { 
+                branch "master" 
+                // evaluate the when condition before entering this stage's agent, if any
+                beforeAgent true 
+            }
             steps {
                 colourText("info", "Building ${env.BUILD_ID} on ${env.JENKINS_URL} from branch ${env.BRANCH_NAME}")
-
-                sh """
-                    $SBT clean compile publishLocal
-                """
+                unstash name: 'Checkout'
+                sh 'sbt package'
+                script {
+                    def uploadSpec = """{
+                        "files": [
+                            {
+                                "pattern": "target/scala-2.10/**/*.jar",
```

Comment:
Should the regex change so that the version version is not hardcoded? If we upgrade to 2.11 the path will fail.